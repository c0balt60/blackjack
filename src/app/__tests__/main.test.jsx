import "@testing-library/jest-dom"
import { render, screen } from "@testing-library/react"
import Game from "@/app/components/game"

describe('MainPage', () => {
    it('renders the game page', () => {
        const {asFragment} = render(<Game/>)

        const player = screen.getByTestId('player', {exact: true})
        const dealer = screen.getByTestId('dealer', {exact: true})

        expect(player).toBeInTheDocument()
        expect(dealer).toBeInTheDocument()
        expect(asFragment()).toMatchSnapshot()
    })
})
