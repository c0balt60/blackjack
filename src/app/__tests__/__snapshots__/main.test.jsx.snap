// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage renders the game page 1`] = `
<DocumentFragment>
  <div
    class="word black"
    data-testid="dealer"
  >
    Dealer:
  </div>
  <div
    class=""
  >
    <span
      class="card black"
    >
      🂮
    </span>
    <span
      class="card black"
    >
      🂠
    </span>
    <div
      class="count black"
    >
      ⇒ 10
    </div>
  </div>
  <div
    class="word black"
    data-testid="player"
  >
    Player $100.00:
  </div>
  <div
    class=""
  >
    <span>
      <span
        class="card black"
      >
        🂩
      </span>
    </span>
    <span>
      <span
        class="card black"
      >
        🃓
      </span>
    </span>
    <div
      class="count black"
    >
      ⇒ 12  $5.00 ⇐  
    </div>
  </div>
  <div
    class="menu-buttons"
  >
    <div
      class="btn btn-dark"
    >
      Double
    </div>
    <div
      class="btn btn-dark"
    >
      Hit
    </div>
    <div
      class="btn btn-dark"
    >
      Stand
    </div>
    <div
      class="btn btn-dark disabled"
    >
      Split
    </div>
  </div>
</DocumentFragment>
`;
